### PR TITLE
Add a separate test for headless mode

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
check headless mode functionality on Xcode9+

### CODE REVIEW

File: test/simulator-e2e-specs.js [hunk 1]
```diff
@@ -260,6 +260,36 @@ function runTests (deviceType) {
       (await sim.hasCalendarAccess(BUNDLE_ID)).should.be.false;
     });
 
+    it('should properly start simulator in headless mode on Xcode9+', async function () {
+      const sim = await getSimulator(udid);
+      if (!(parseInt(sim.constructor.name.replace(/\D/g, ''), 10) >= 9)) {
```
Comment (mykola-mokhnach) [inline test/simulator-e2e-specs.js]: Is there a better way to detect the current xcode version here?

Comment (imurchie) [inline test/simulator-e2e-specs.js]: Yes. We already have `appium-xcode` as a dependency. So you can just use something like:
```js
import xcode from 'appium-xcode';

// ...

before(async function () {
  let xcodeVersion = await xcode.getVersion(true);
  if (xcodeVersion.major < 9) return this.skip();
});
```

File: test/simulator-e2e-specs.js [hunk 2]
```diff
@@ -260,6 +261,36 @@ function runTests (deviceType) {
       (await sim.hasCalendarAccess(BUNDLE_ID)).should.be.false;
     });
 
+    it('should properly start simulator in headless mode on Xcode9+', async function () {
+      const sim = await getSimulator(udid);
```
Comment (imurchie) [inline test/simulator-e2e-specs.js]: These should be revered. There is no reason to have the sim object if we are going to skip.

**Comment by mykola-mokhnach** [comment]
@imurchie done. Also moved verification method to the top, because I'm going to reuse it for parallel session tests (https://github.com/appium/appium-ios-simulator/pull/123)

**Comment by imurchie** [comment]
Awesome! Yes. I noted that on the other PR :)

**Pull request merged**